src/main/ruby/core/dir_glob.rb

```diff
@@ -26,6 +29,10 @@
 
 class Dir
   module Glob
+    no_meta_chars = '[^*?\\[\\]{}\\\\]'
```

Comment:
This can also be a Regexp literal to avoid the doubled `\`, but this is fine too.